src/util/child_common.c

```diff
@@ -521,8 +521,23 @@ static void _read_pipe_handler(struct tevent_context *ev,
         return;
     }
 
+    buf = talloc_array(state, uint8_t, CHILD_MSG_CHUNK);
+    if (buf == NULL) {
+        tevent_req_error(req, ENOMEM);
+        return;
+    }
+
     if (state->safe) {
         size = sss_atomic_read_safe_s(state->fd, buf, CHILD_MSG_CHUNK, &len);
+        if (size == -1 && errno == ERANGE) {
```

Comment:
Would it make sense to make `k5c_recv_data()` aware of this specific error code as well? (probably not, asking just in case)

Reply:
I asked Sumit about this earlier, he thinks it is okay to just error out in this case.

Reply:
Ok, thanks.